pyccel/codegen/printing/ccode.py

```diff
@@ -989,16 +1003,28 @@ def get_print_format_and_arg(self, var):
             The code which should be printed in the arguments of the generated
             print expression to print the object.
         """
-        try:
-            arg_format = self.type_to_format[(var.dtype, get_final_precision(var))]
-        except KeyError:
-            errors.report("{} type is not supported currently".format(var.dtype), severity='fatal')
-        if var.dtype is NativeComplex():
-            arg = '{}, {}'.format(self._print(NumpyReal(var)), self._print(NumpyImag(var)))
-        elif var.dtype is NativeBool():
-            arg = '{} ? "True" : "False"'.format(self._print(var))
+        if isinstance(var.dtype, FixedSizeNumericType):
+            primitive_type = var.dtype.primitive_type
+            if isinstance(primitive_type, PrimitiveComplexType):
+                _, real_part = self.get_print_format_and_arg(NumpyReal(var))
+                float_format, imag_part = self.get_print_format_and_arg(NumpyImag(var))
+                return f'({float_format} + {float_format}j)', f'{real_part}, {imag_part}'
+            elif isinstance(primitive_type, PrimitiveBooleanType):
+                return self.get_print_format_and_arg(IfTernaryOperator(var, LiteralString("True"), LiteralString("False")))
```

Comment:
Are more cases covered now? The original function was not recursive...

Reply:
This should be the the same as before. I had to rewrite it to fix the f-string problems and to use the precision to search in `type_to_format` (as only `FixedSizeNumericType`s have this property).

Reply:
I use recursivity to try to reduce the number of assumptions made hopefully making it easier to maintain